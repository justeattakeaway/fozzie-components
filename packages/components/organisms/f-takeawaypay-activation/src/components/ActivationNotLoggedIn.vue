<template>
    <div>
        <bag-celebrate-bg-icon :class="$style['c-takeawaypayActivation-icon']" />

        <h1 :class="$style['c-takeawaypayActivation-title']">
            {{ $t('messages.titleActivationNotLoggedIn') }}
        </h1>

        <p>
            {{ $t('messages.descriptionActivationNotLoggedIn') }}
        </p>

        <div :class="$style['c-takeawaypayActivation-actions']">
            <f-button
                :href="loginUrl"
                data-test-id="takeawaypay-activation-not-loggedin-login-link"
                button-type="primary"
                button-size="large"
                is-full-width>
                {{ $t('actions.loginToAccount2') }}
            </f-button>

            <f-button
                :href="registrationUrl"
                data-test-id="takeawaypay-activation-not-loggedin-registration-link"
                button-type="outline"
                button-size="large"
                is-full-width>
                {{ $t('actions.createAccount') }}
            </f-button>
        </div>
    </div>
</template>

<script>
import FButton from '@justeat/f-button';
import '@justeat/f-button/dist/f-button.css';
import { BagCelebrateBgIcon } from '@justeat/f-vue-icons';

export default {

    components: {
        FButton,
        BagCelebrateBgIcon
    },

    props: {
        loginUrl: {
            type: String,
            required: true
        },
        registrationUrl: {
            type: String,
            required: true
        }
    }
};
</script>

<style lang="scss" module>

.c-takeawaypayActivation-icon {
    transform: translate(10%);
    width: 125px;
}

.c-takeawaypayActivation-title {
    @include font-size(body-l);
    font-weight: $font-weight-bold;
}

.c-takeawaypayActivation-actions {
    padding-top: spacing(x3);
}

</style>
