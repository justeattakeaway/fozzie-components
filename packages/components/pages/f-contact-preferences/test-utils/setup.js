export const contactPreferencesGetResponse = {
    data: {
        Preferences: [{
            DisplayName: 'Order status',
            Sort: 1,
            Key: 'orderstatus',
            Push: false,
            Email: true,
            Sms: true
        }, {
            DisplayName: 'Review meal',
            Sort: 2,
            Key: 'reviewmeal',
            Push: false,
            Email: false,
            Sms: false
        }, {
            DisplayName: 'News & offers',
            Sort: 3,
            Key: 'news',
            Push: false,
            Email: false,
            Sms: false
        }
        ],
        PreferenceVersionViewed: 0
    }
};

export const contactPreferencesViewModel = {
    preferenceVersionViewed: 0,
    preferences: [{
        displayName: 'Order status',
        isEmailEnabled: false,
        emailValue: true,
        key: 'orderstatus',
        pushEnabled: false,
        pushValue: false,
        isSmsEnabled: false,
        smsValue: true,
        sort: 1,
        visible: true
    }, {
        displayName: 'Review meal',
        isEmailEnabled: true,
        emailValue: false,
        key: 'reviewmeal',
        pushEnabled: true,
        pushValue: false,
        isSmsEnabled: true,
        smsValue: false,
        sort: 2,
        visible: false
    }, {
        displayName: 'News & offers',
        isEmailEnabled: true,
        emailValue: false,
        key: 'news',
        pushEnabled: true,
        pushValue: false,
        isSmsEnabled: true,
        smsValue: false,
        sort: 3,
        visible: true
    }]
};

export const filteredContactPreferencesModel = [{
    displayName: 'Order status',
    isEmailEnabled: false,
    emailValue: true,
    key: 'orderstatus',
    pushEnabled: false,
    pushValue: false,
    isSmsEnabled: false,
    smsValue: true,
    sort: 1,
    visible: true
}, {
    displayName: 'News & offers',
    isEmailEnabled: true,
    emailValue: false,
    key: 'news',
    pushEnabled: true,
    pushValue: false,
    isSmsEnabled: true,
    smsValue: false,
    sort: 3,
    visible: true
}];

export const contactPreferencesUpdateModel = {
    Preferences: [{
        DisplayName: 'Order status',
        Sort: 1,
        Key: 'orderstatus',
        Push: false,
        Email: true,
        Sms: true
    }, {
        DisplayName: 'Review meal',
        Sort: 2,
        Key: 'reviewmeal',
        Push: false,
        Email: false,
        Sms: false
    }, {
        DisplayName: 'News & offers',
        Sort: 3,
        Key: 'news',
        Push: false,
        Email: false,
        Sms: false
    }],
    DeviceToken: null,
    DeviceType: null,
    PhoneNumber: null,
    PreferenceVersionViewed: 0
};

export const baseUrl = 'https://smartGatewayBaseUrl.com';

export const token = 'some-auth-token';

export const conversationId = 'b7386108-95e6-4e73-9421-5b066c089153';

export const contactPreferencesUpdateBody = {
    Preference: [{
        DisplayName: 'Order status',
        Sort: 1,
        Key: 'orderstatus',
        Push: false,
        Email: true,
        Sms: true
    }, {
        DisplayName: 'Review meal',
        Sort: 2,
        Key: 'reviewmeal',
        Push: false,
        Email: false,
        Sms: false
    }, {
        DisplayName: 'News & offers',
        Sort: 3,
        Key: 'news',
        Push: false,
        Email: true,
        Sms: false
    }
    ],
    DeviceToken: null,
    DeviceType: null,
    PhoneNumber: null,
    PreferenceVersionViewed: 0
};
