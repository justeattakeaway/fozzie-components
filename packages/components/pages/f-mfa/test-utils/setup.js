export const baseUrl = 'https://smartGatewayBaseUrl.com';

export const conversationId = 'b7386108-95e6-4e73-9421-5b066c089153';

// eslint-disable-next-line camelcase
export const postBody = { mfa_token: 'some_mfa_token', otp: 'ABC123' };
