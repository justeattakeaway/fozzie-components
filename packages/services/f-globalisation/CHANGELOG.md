# Changelog

The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).

v1.0.0
------------------------------
*December 17, 2020*

### Added
- Support for `dateTimeFormats`, which requires a change in the structure of the tenancy files.


v0.2.0
------------------------------
*November 10, 2020*

### Added
- Provided a wrapper around the vue-i18n module for component tests to import


v0.1.0
------------------------------
*November 9, 2020*

### Added
- Added a Globalisation Mixin which performs all required wiring to integrate vue-i18n
