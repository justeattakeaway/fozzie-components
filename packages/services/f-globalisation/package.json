{
  "name": "@justeat/f-globalisation",
  "description": "Fozzie Globalisation â€“ A utility which wires up vue-i18n within your Smart Component",
  "version": "1.1.0",
  "main": "dist/f-globalisation.umd.min.js",
  "maxBundleSize": "15kB",
  "files": [
    "dist"
  ],
  "publishConfig": {
    "access": "public"
  },
  "homepage": "https://github.com/justeat/fozzie-components/tree/master/packages/services/f-globalisation",
  "contributors": [
    "Github contributors <https://github.com/justeat/fozzie-components/graphs/contributors>"
  ],
  "repository": {
    "type": "git",
    "url": "git@github.com:justeat/fozzie-components.git"
  },
  "bugs": {
    "url": "https://github.com/justeat/fozzie-components/issues"
  },
  "license": "Apache-2.0",
  "engines": {
    "node": "^12 || ^14 || ^16"
  },
  "keywords": [
    "fozzie"
  ],
  "scripts": {
    "prepublishOnly": "yarn lint && yarn test && yarn build",
    "build": "vue-cli-service build --target lib --name f-globalisation ./src/index.js",
    "lint": "eslint \"!(dist)/**/*.{js,vue}\"",
    "lint:fix": "yarn lint --fix",
    "test": "jest",
    "ci:test:services": "yarn test"
  },
  "browserslist": [
    "extends @justeat/browserslist-config-fozzie"
  ],
  "dependencies": {
    "vue-i18n": "8.28.2"
  }
}
