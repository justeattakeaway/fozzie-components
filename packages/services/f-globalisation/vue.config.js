module.exports = {
    pluginOptions: {
        lintStyleOnBuild: true
    }
};
