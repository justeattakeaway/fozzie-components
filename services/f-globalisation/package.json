{
  "name": "@justeat/f-globalisation",
  "description": "Fozzie Globalisation â€“ A utility which wires up vue-i18n within your Smart Component",
  "version": "0.2.0",
  "main": "dist/f-globalisation.umd.min.js",
  "files": [
    "dist"
  ],
  "homepage": "https://github.com/justeat/fozzie-components/tree/master/packages/f-globalisation",
  "contributors": [
    "Github contributors <https://github.com/justeat/fozzie-components/graphs/contributors>"
  ],
  "repository": {
    "type": "git",
    "url": "git@github.com:justeat/fozzie-components.git"
  },
  "bugs": {
    "url": "https://github.com/justeat/fozzie-components/issues"
  },
  "license": "Apache-2.0",
  "engines": {
    "node": ">=10.0.0"
  },
  "keywords": [
    "fozzie"
  ],
  "scripts": {
    "prepublishOnly": "yarn lint && yarn test && yarn build",
    "build": "vue-cli-service build --target lib --name f-globalisation ./src/index.js",
    "lint": "vue-cli-service lint",
    "lint:fix": "yarn lint --fix",
    "test": "vue-cli-service test:unit --json"
  },
  "browserslist": [
    "extends @justeat/browserslist-config-fozzie"
  ],
  "dependencies": {
    "vue-i18n": "8.0.0"
  },
  "devDependencies": {
    "@samhammer/vue-cli-plugin-stylelint": "2.0.1",
    "@vue/cli-plugin-babel": "4.4.6",
    "@vue/cli-plugin-eslint": "3.9.2",
    "@vue/cli-plugin-unit-jest": "4.4.6",
    "@vue/test-utils": "1.0.3",
    "vue": "2.6.10"
  }
}
