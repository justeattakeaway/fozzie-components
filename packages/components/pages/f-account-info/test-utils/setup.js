import Vuex from 'vuex';
import { createLocalVue } from '@vue/test-utils';
import { VueI18n } from '@justeat/f-globalisation';
import tenantConfigs from '../src/tenants';

export const localVue = createLocalVue();
localVue.use(VueI18n);
localVue.use(Vuex);

export const i18n = {
    locale: 'en-GB',
    messages: {
        'en-GB': tenantConfigs['en-GB'].messages
    }
};

export const baseUrl = 'https://smartGatewayBaseUrl.com';

export const token = 'some-auth-token';

export const conversationId = 'b7386108-95e6-4e73-9421-5b066c089153';

export const consumerDetailsGetResponse = {
    data: {
        ConsumerId: 1050450174,
        Id: '8It2lEaJjG82nJtxO00Iab0CoF0=',
        CreatedDate: '2019-04-16T17:25:00.000Z',
        EmailAddress: 'satchin.moriyoka@example.com',
        FirstName: 'Satchin',
        LastName: 'Moriyoka',
        PhoneNumber: '+441234567890',
        DateOfBirth: '1914-04-16T17:25:00.000Z',
        TermsAndConditionsId: 0,
        WantsNewsletter: true,
        AccountType: 'Default',
        ConsumerStatus: 'ReturnConsumer'
    }
};

const consumerAddressGetResponse = {
    AddressId: 1050450174,
    City: 'Bristol',
    ZipCode: 'BS1 4DJ',
    AddressName: 'Office',
    IsDefault: true, // Key field
    Line1: 'Just Eat',
    Line2: 'Broad Quay House'
};

export const consumerAddressesGetResponse = {
    data: [{
        ...consumerAddressGetResponse
    },
    {
        AddressId: 1050450175,
        City: 'Bristol',
        ZipCode: 'BS1 9DJ',
        AddressName: 'Office',
        IsDefault: false,
        Line1: 'Takeaway',
        Line2: 'Broad Quay House'
    }]
};

export const consumerDetailsMappedModel = {
    id: consumerDetailsGetResponse.data.ConsumerId,
    firstName: consumerDetailsGetResponse.data.FirstName,
    lastName: consumerDetailsGetResponse.data.LastName,
    phoneNumber: consumerDetailsGetResponse.data.PhoneNumber,
    emailAddress: consumerDetailsGetResponse.data.EmailAddress
};

export const consumerAddressMappedModel = {
    line1: consumerAddressGetResponse.Line1,
    line2: consumerAddressGetResponse.Line2,
    line3: consumerAddressGetResponse.Line3,
    locality: consumerAddressGetResponse.City,
    postcode: consumerAddressGetResponse.ZipCode
};

export const consumerViewModel = {
    ...consumerDetailsMappedModel,
    ...consumerAddressMappedModel
};

export const consumerStateModel = {
    consumer: {
        ...consumerDetailsMappedModel,
        ...consumerAddressMappedModel
    }
};
