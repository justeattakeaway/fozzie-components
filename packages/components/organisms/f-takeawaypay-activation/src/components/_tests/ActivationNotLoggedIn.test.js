import { shallowMount, createLocalVue } from '@vue/test-utils';
import { VueI18n } from '@justeat/f-globalisation';
import ActivationNotLoggedIn from '../ActivationNotLoggedIn.vue';
import i18n from './helpers/setup';

const localVue = createLocalVue();
localVue.use(VueI18n);

describe('ActivationNotLoggedIn', () => {
    const propsData = {
        loginUrl: '/login',
        registrationUrl: '/registration',
        locale: 'en-GB'
    };

    const wrapper = shallowMount(ActivationNotLoggedIn, {
        i18n,
        localVue,
        propsData
    });

    it('should be defined', () => {
        expect(wrapper.exists()).toBe(true);
    });

    it('should be rendered correctly', () => {
        const container = wrapper.find('[data-test-id="activation-not-logged-in-component"]');

        expect(container).toMatchSnapshot();
    });
});
