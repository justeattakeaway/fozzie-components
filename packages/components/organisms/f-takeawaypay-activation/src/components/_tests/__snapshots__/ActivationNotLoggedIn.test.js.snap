// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivationNotLoggedIn should be rendered correctly 1`] = `
<div data-test-id="activation-not-logged-in-component" locale="en-GB">
  <bag-celebrate-bg-icon-stub></bag-celebrate-bg-icon-stub>
  <h1>
    Let's get you started with Takeaway Pay
  </h1>
  <p>
    Do you already have a Just Eat account?
  </p>
  <div>
    <f-button-stub buttontype="primary" buttonsize="large" isfullwidth="true" actiontype="button" href="/login" data-test-id="takeawaypay-activation-not-loggedin-login-link">
      I have an account
    </f-button-stub>
    <f-button-stub buttontype="outline" buttonsize="large" isfullwidth="true" actiontype="button" href="/registration" data-test-id="takeawaypay-activation-not-loggedin-registration-link">
      Create a new account
    </f-button-stub>
  </div>
</div>
`;
